### ISSUE TITLE
Password generator generates passwords that have an invalid format

### ISSUE THREAD

**Issue by OrangeJuiced** [Pterodactyl/Panel#191]
The password generator in the panel sometimes generates passwords that do not comply with the rules set for the password format. Example password generated that doesn't work: TbMbtjliTfRJ

**Comment by OrangeJuiced** [comment]
I hotfixed this using my basic PHP knowledge.
```
    public function getPassword(Request $request, $length = 16)
    {
		$gensuccess =  false;
		
		while($gensuccess == false)
		{
			$length = ($length < 8) ? 8 : $length;
			$genpw = str_random($length);			
			
			if (preg_match('/[A-Z]+[a-z]+[0-9]+/', $genpw))
			{	
				$gensuccess = true;
				return $genpw;					
			}				
		}	
    }
```

There's probably a better way I don't know of to do this though.

**Comment by DaneEveritt** [comment]
closed in ed5b755
